client/src/components/DisplayTable/DisplayTable.js

```diff
@@ -266,6 +280,62 @@ export class DisplayTable extends React.Component {
       .reverse()
       .value();
 
+    const {
+      paginate_by,
+      paginated_data,
+      change_paginate_by,
+      show_pagination_controls,
+      page_selector,
+    } = (() => {
+      const paginate_by =
+        size_to_paginate && _.size(sorted_filtered_data) > size_to_paginate
+          ? state_paginate_by
+          : 0;
+
+      const num_pages_exact = paginate_by
+        ? _.size(sorted_filtered_data) / paginate_by
+        : -1;
+
+      const num_pages =
+        num_pages_exact >= 0 && _.isInteger(num_pages_exact) //if we have the exact amount of items to fill an entire page, it will start a new empty page
+          ? num_pages_exact - 1 //thus we want to get rid of the extra page (as it has no content)
+          : _.toInteger(num_pages_exact); //start counting from 0 (-1 from the literal amount of pages)
+
+      const paginated_data = _.chunk(
+        sorted_filtered_data,
+        paginate_by || _.size(sorted_filtered_data)
+      )[current_page];
+
+      const change_page = (page) => {
+        this.setState({ current_page: page });
+        this.first_data_ref.current.focus();
+      };
+
+      const change_paginate_by = (new_paginate_by) => {
+        this.setState({ state_paginate_by: new_paginate_by, current_page: 0 });
+      };
+
+      const show_pagination_controls = num_pages >= 0;
+
+      const page_selector = show_pagination_controls && (
+        <PageSelector
+          num_pages={num_pages}
+          current_page={current_page}
+          change_page={change_page}
+          show_select
+          num_col={_.size(visible_ordered_col_keys)}
+        />
+      );
```

Comment:
I'd suggest putting `PageinateBySelector` in the same row as `PageSelector`. They share the same requirement (`show_pagination_controls`) and are tightly coupled conceptually.